packages/ui-react/src/components/LanguageMenu/LanguageMenu.test.tsx

```diff
@@ -1,44 +1,56 @@
+import type { LanguageDefinition } from '@jwp/ott-common/types/i18n';
 
 import LanguageMenu from './LanguageMenu';
 
 const languages = [
-  {
-    displayName: 'English',
-    code: 'en',
-  },
-  {
-    displayName: 'español',
-    code: 'es',
-  },
+  { displayName: 'English', code: 'en' },
+  { displayName: 'español', code: 'es' },
 ];
 
-describe('<LanguageMenu>', () => {
-  test('renders and matches snapshot', () => {
-    const currentLanguage = languages[1];
-    const { container } = renderWithRouter(<LanguageMenu languages={languages} currentLanguage={currentLanguage} onClick={() => undefined} />);
+const onClickCb = vi.fn();
+const openLanguageMenuCb = vi.fn();
+const closeLanguageMenuCb = vi.fn();
 
-    expect(container).toMatchSnapshot();
+describe('<LanguageMenu>', () => {
+  afterEach(() => {
+    onClickCb.mockClear();
+    openLanguageMenuCb.mockClear();
+    closeLanguageMenuCb.mockClear();
+    cleanup();
   });
 
-  test('shows all languages in the menu', () => {
-    const currentLanguage = languages[1];
-    const { queryByText } = renderWithRouter(<LanguageMenu languages={languages} currentLanguage={currentLanguage} onClick={() => undefined} />);
+  const renderLanguageMenu = (currentLanguage: LanguageDefinition, languageMenuOpen: boolean) => {
+    return render(
+      <LanguageMenu
+        languages={languages}
+        currentLanguage={currentLanguage}
+        onClick={onClickCb}
+        languageMenuOpen={languageMenuOpen}
+        openLanguageMenu={openLanguageMenuCb}
+        closeLanguageMenu={closeLanguageMenuCb}
+      />,
+    );
+  };
 
-    expect(queryByText('English')).toBeInTheDocument();
-    expect(queryByText('español')).toBeInTheDocument();
+  test('renders and matches snapshot', () => {
+    const { container } = renderLanguageMenu(languages[0], false);
+    expect(container).toMatchSnapshot();
   });
 
-  test('renders languages and calls the onClick callback with the correct language code', () => {
-    const currentLanguage = languages[1];
-    const callback = vi.fn();
-    const { getByText } = renderWithRouter(<LanguageMenu languages={languages} currentLanguage={currentLanguage} onClick={callback} />);
+  test('Renders the correct items in the LanguageMenu', () => {
+    const { container, queryByText } = renderLanguageMenu(languages[0], true);
+    const languagePanel = container.querySelector('#language-panel');
+    expect(languagePanel).toBeInTheDocument();
 
-    fireEvent.click(getByText('English'));
+    expect(queryByText(languages[0].displayName)).toBeInTheDocument();
+    expect(queryByText(languages[1].displayName)).toBeInTheDocument();
+  });
 
-    expect(callback).toBeCalledWith('en');
+  test('Check if the correct language item is selected onclick', () => {
+    const { container } = renderLanguageMenu(languages[0], true);
+    const languageMenuItems = container.querySelectorAll('#language-panel li');
+    fireEvent.click(languageMenuItems[0]);
```

Comment:
```suggestion
    fireEvent.click(getByText(languages[0].displayName));
```
This approach should be more appropriate